processor/src/main/java/org/mapstruct/ap/internal/model/source/ForgedMethod.java

```diff
@@ -303,4 +306,14 @@ public int hashCode() {
         result = 31 * result + ( name != null ? name.hashCode() : 0 );
         return result;
     }
+
+    private Parameter determineMappingTargetParameter(Iterable<Parameter> parameters) {
```

Comment:
I could do it as static on Parameter. Just like @agudian did for context parameters as well.